apitools/base/py/encoding.py

```diff
@@ -710,3 +744,68 @@ def _IsRepeatedJsonValue(msg):
     if isinstance(msg, extra_types.JsonArray):
         msg = msg.entries
     return msg
+
+
+def _IsMap(message, field):
+  """Returns whether the "field" is actually a map-type."""
+  value = message.get_assigned_value(field.name)
+  if not isinstance(value, messages.Message):
+    return False
+  try:
+    additional_properties = value.field_by_name('additionalProperties')
+  except KeyError:
+    return False
+  else:
+    return additional_properties.repeated
+
+
+def _MapItems(message, field):
+  """Yields the (key, value) pair of the map values."""
+  assert _IsMap(message, field)
+  map_message = message.get_assigned_value(field.name)
+  additional_properties = map_message.get_assigned_value('additionalProperties')
+  for kv_pair in additional_properties:
+    yield kv_pair.key, kv_pair.value
+
+
+def UnrecognizedFieldIter(message, _edges=()):  # pylint: disable=invalid-name
+  """Yields (edges_to_error, unused_fields) for sub-messages with unused fields.
```

Comment:
Mismatch between the brief description and the Yields section

Reply:
Done.